x-pack/plugins/apm/public/components/app/ServiceMap/Cytoscape.tsx

```diff
@@ -134,6 +134,11 @@ export function Cytoscape({
       );
       cy.remove(absentElements);
       cy.add(elements);
+      // ensure all elements get latest data properties
+      elements.forEach(elementDefinition => {
+        const el = cy.getElementById(elementDefinition.data.id as string);
+        el.data(elementDefinition.data);
+      });
```

Comment:
This was necessary to add since the `data` of each node wasn't being updated when existing elements were re-added. Before adding this, the displayed anomaly data was stale.